packages/at_secondary_server/lib/src/verb/handler/enroll_verb_handler.dart

```diff
@@ -143,13 +156,39 @@ class EnrollVerbHandler extends AbstractVerbHandler {
       enrollDataStoreValue.approval!.state =
           _getEnrollStatusEnum(operation).name;
       responseJson['status'] = _getEnrollStatusEnum(operation).name;
-
       AtData updatedEnrollData = AtData()
         ..data = jsonEncode(enrollDataStoreValue.toJson());
-
-      await keyStore.put('$key$currentAtSign', updatedEnrollData);
+      await keyStore.put('$key$currentAtSign', updatedEnrollData,
+          skipCommit: true);
+      // when enrollment is approved store the apkamPublicKey of the enrollment
+      if (operation == 'approve') {
+        var apkamPublicKeyInKeyStore =
+            'public:${enrollDataStoreValue.appName}.${enrollDataStoreValue.deviceName}.pkam.$pkamNamespace.__public_keys$currentAtSign';
+        var valueJson = {};
+        valueJson[apkamPublicKey] = enrollDataStoreValue.apkamPublicKey;
+        var atData = AtData()..data = jsonEncode(valueJson);
+        await keyStore.put(apkamPublicKeyInKeyStore, atData);
+        await _storeEncryptionKeys(
+            enrollmentIdFromParams!, enrollParams, currentAtSign);
+      }
     }
-    responseJson['enrollmentId'] = enrollmentId;
+    responseJson['enrollmentId'] = enrollmentIdFromParams;
+  }
+
+  Future<void> _storeEncryptionKeys(
+      String newEnrollmentId, EnrollParams enrollParams, String atSign) async {
+    var privKeyJson = {};
+    privKeyJson['value'] = enrollParams.encryptedDefaultEncryptedPrivateKey;
+    await keyStore.put(
+        '$newEnrollmentId.$defaultEncryptionPrivateKey.$enrollManageNamespace$atSign',
+        AtData()..data = jsonEncode(privKeyJson),
+        skipCommit: true);
+    var selfKeyJson = {};
+    selfKeyJson['value'] = enrollParams.encryptedDefaultSelfEncryptionKey;
+    await keyStore.put(
+        '$newEnrollmentId.$defaultSelfEncryptionKey.$enrollManageNamespace$atSign',
+        AtData()..data = jsonEncode(selfKeyJson),
+        skipCommit: true);
   }
 
   EnrollStatus _getEnrollStatusEnum(String? enrollmentOperation) {
```

Comment:
Given that the comparisons below are to lowercase, please coerce the enrollmentOperation to lower case if it is non-null, before entering the switch

Reply:
Also would be really good to not use a switch statement but instead to use the enum's `values.byName()` method.

The equivalent of this method in one line of code would be something like
`EnrollStatus.values.byName((enrollmentOperation ?? 'pending').toLowerCase());`

Reply:
Enum values and the operation name are not identical.  Hence byName will not work in this case.
e.g approved - approve, denied- deny, revoked-revoke.
I have removed switch case and simplified the code
https://github.com/atsign-foundation/at_server/pull/1496/commits/b22ea2028164aa54299758080d3b9033bf3a98e2